doctr/models/classification/vit/pytorch.py

```diff
@@ -58,26 +80,15 @@ def __init__(
         cfg: Optional[Dict[str, Any]] = None,
     ) -> None:
 
-        super().__init__()
-        self.cfg = cfg
-        self.include_top = include_top
-
-        self.patch_embedding = PatchEmbedding(input_shape, patch_size, d_model)
-        self.encoder = EncoderBlock(num_layers, num_heads, d_model, dropout, nn.GELU())
-
-        if self.include_top:
-            self.head = nn.Linear(d_model, num_classes)
+        _layers: List[nn.Module] = [
+            PatchEmbedding(input_shape, patch_size, d_model),
+            EncoderBlock(num_layers, num_heads, d_model, dropout, nn.GELU()),
+        ]
+        if include_top:
+            _layers.append(ClassifierHead(d_model, num_classes))
```

Comment:
Sorry, I forgot to ask it in the first review: why is it enough to do `forward` on the first dimension of `x` in `ClassifierHeader` ?

Reply:
in PatchEmbedding we add a cls_token at the first position which is used for classification the other are features

Reply:
Thanks